packages/training/python/src/training/multi_turn.py

```
@@ -0,0 +1,656 @@
+            self._finalize()
+    
+    def _finalize(self) -> None:
+        """Finalize episode after completion"""
+        self.completed = True
+        self.end_time = datetime.now(timezone.utc)
+        self.episode_length = len(self.turns)
+        self.total_reward = sum(t.reward for t in self.turns)
+        
+        # Check success based on final reward
+        if self.turns:
+            final_turn = self.turns[-1]
```

Ruff (0.14.10)

Local variable `final_turn` is assigned to but never used

Remove assignment to unused variable `final_turn`

(F841)